File: 2582-pass-the-pillow/2582-pass-the-pillow.cpp
class Solution {
public:
    int passThePillow(int n, int time) {
        int cur = 1;
        bool dir = true;
        for (int t = 1; t <= time; t++) {
            if (dir) {
                cur++;
            } else {
                cur--;
            }
            if (cur == 1 || cur == n) {
                dir = !dir;
            }
        }
        return cur;
    }
};

File: 2582-pass-the-pillow/2582-pass-the-pillow.java
class Solution {
    public int passThePillow(int n, int time) {
        int pillow = 1, dir = 1;
        for (int i = 1; i <= time; i++) {
            pillow += dir;
            if (pillow == 1 || pillow == n) {
                dir *= -1;
            }
        }
        return pillow;
    }
}